build-tools/packages/version-tools/src/internalVersionScheme.ts

```diff
@@ -497,3 +497,47 @@ export function detectInternalVersionConstraintType(
 	const maxSatisfying = semver.maxSatisfying([patch, minor], range);
 	return maxSatisfying === patch ? "patch" : maxSatisfying === minor ? "minor" : "exact";
 }
+
+/**
+ * Checks if the provided version is a test version.
+ *
+ * Test versions are generated from test/ branches and are published to the test feed.
+ *
+ * @param version - The version to check
+ * @returns - True if the version string is a test version, otherwise false
+ *
+ * @example
+ * returns true
+ * isInternalTestVersion("0.0.0-260312-test");
+ *
+ * @example
+ * returns false
+ * isInternalTestVersion("2.1.0-260312");
+ */
+export function isInternalTestVersion(version: semver.SemVer | string): boolean {
```

Comment:
I know Tyler suggested taking both string/Semver, but I'd typically recommend only taking one or the other, and preferring the more-strongly-typed option (so only taking Semver here).

This then pushes the type to be made stronger earlier in the callstack and let others take advantage of that strong typing.  E.g. extractBuildNumber can call split on the prerelease, rather than calling it on the full version string.  Taken to the extreme, you could push it all the way up to UnreleaseReportCommand.run() such that SemVer is used all through the stack for maximum strong typing.

Up to you for how you want to resolve, but wanted to offer the alternate opinion.

Reply:
I agree with this in principal, but in practice it ends up being pretty unwieldy imo, and it's out of scope for this change. It also requires everything to take a deep on the sender package in order to use this one. Good debate to have separately though.